bin/host/src/handler.rs

```diff
@@ -0,0 +1,184 @@
+use alloy_primitives::keccak256;
+//use alloy_primitives::B256;
+use alloy_rlp::Decodable;
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use hokulea_proof::hint::ExtendedHintType;
+use kona_host::SharedKeyValueStore;
+use kona_host::{single::SingleChainHintHandler, HintHandler, OnlineHostBackendCfg};
+
+use crate::cfg::SingleChainHostWithEigenDA;
+
+use kona_preimage::{PreimageKey, PreimageKeyType};
+use kona_proof::Hint;
+use tracing::trace;
+
+use eigenda_v2_struct_rust::EigenDAV2Cert;
+//use hokulea_compute_kzg_proof::compute_kzg_proof;
+use hokulea_eigenda::BlobInfo;
+use hokulea_eigenda::EigenDABlobData;
+use hokulea_eigenda::{BYTES_PER_FIELD_ELEMENT, PAYLOAD_ENCODING_VERSION_0};
+
+/// The [HintHandler] for the [SingleChainHostWithEigenDA].
+#[derive(Debug, Clone, Copy)]
+pub struct SingleChainHintHandlerWithEigenDA;
+
+#[async_trait]
+impl HintHandler for SingleChainHintHandlerWithEigenDA {
+    type Cfg = SingleChainHostWithEigenDA;
+
+    /// A wrapper that route eigenda hint and kona hint
+    async fn fetch_hint(
+        hint: Hint<<Self::Cfg as OnlineHostBackendCfg>::HintType>,
+        cfg: &Self::Cfg,
+        providers: &<Self::Cfg as OnlineHostBackendCfg>::Providers,
+        kv: SharedKeyValueStore,
+    ) -> Result<()> {
+        // route the hint to the right fetcher based on the hint type.
+        match hint.ty {
+            ExtendedHintType::EigenDACertV1 | ExtendedHintType::EigenDACertV2 => {
+                fetch_eigenda_hint(hint, cfg, providers, kv).await?;
+            }
+            ExtendedHintType::Original(ty) => {
+                let hint_original = Hint {
+                    ty,
+                    data: hint.data,
+                };
+                SingleChainHintHandler::fetch_hint(
+                    hint_original,
+                    &cfg.kona_cfg,
+                    &providers.kona_providers,
+                    kv,
+                )
+                .await?;
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Fetch the preimage for the given hint and insert it into the key-value store.
+pub async fn fetch_eigenda_hint(
```

Comment:
should we make a separate function for v1 and v2?

Especially when we add the commitment logic, those will be separate for v1 and v2 right? So we'll have 2 switches in here? In general I prefer to #move-ifs-up as much as possible ;)